### PR TITLE
Allow single buildpack in multiple groups

### LINKED ISSUE #858: --buildpack flag is throwing incorrect version error when buildpack is listed twice in order
### Summary
When I specify a buildpack by ID with the `--buildpack` flag during `pack build`, `pack` returns the following error stating that an explicit version is required because there are multiple version of the buildpack on the builder image.
```
ERROR: failed to build: processing order: unable to resolve version: multiple versions of paketo-buildpacks/image-labels - must specify an explicit version
```
There is only one version of the given buildpack on the builder image. However, the buildpack is included in two separate _groups_. My guess is this is the source of the bug.

---

##### Steps

1. Fetch the Paketo [sample apps](https://github.com/paketo-buildpacks/samples/tree/main/nodejs/yarn) (or any app at all really).
```
git clone https://github.com/paketo-buildpacks/samples
cd samples
```
2. Build with the suggested paketo builder and explicitly include the `paketo-buildpacks/image-labels` or `paketo-buildpacks/procfile` buildpack.
```
pack build samples/nodejs \
  --path nodejs/yarn \
  --builder paketobuildpacks/builder:base \
  --buildpack paketo-buildpacks/nodejs \
  --buildpack paketo-buildpacks/image-labels \
  --env "BP_OCI_DESCRIPTION=Sample App"
```

##### Current behavior

`pack` returns an error and tells me to specify the version of the `paketo-buildpacks/image-labels` buildpack.

##### Expected behavior

It should succeed given that there is exactly one version of `paketo-buildpacks/image-labels` on the builder image.
```
pack inspect-builder paketobuildpacks/builder:base
...
Buildpacks:
  ID                                                  VERSION        HOMEPAGE
  paketo-buildpacks/dotnet-core                       0.0.8          https://github.com/paketo-buildpacks/dotnet-core
  paketo-buildpacks/icu                               0.0.97         https://github.com/paketo-buildpacks/icu
  paketo-buildpacks/node-engine                       0.0.262        https://github.com/paketo-buildpacks/node-engine
  paketo-buildpacks/dotnet-core-aspnet                0.0.190        https://github.com/paketo-buildpacks/dotnet-core-aspnet
  paketo-buildpacks/dotnet-core-build                 0.0.119        https://github.com/paketo-buildpacks/dotnet-core-build
  paketo-buildpacks/dotnet-core-conf                  0.0.177        https://github.com/paketo-buildpacks/dotnet-core-conf
  paketo-buildpacks/dotnet-core-runtime               0.0.195        https://github.com/paketo-buildpacks/dotnet-core-runtime
  paketo-buildpacks/dotnet-core-sdk                   0.0.192        https://github.com/paketo-buildpacks/dotnet-core-sdk
  paketo-buildpacks/java-native-image                 3.3.2          https://github.com/paketo-buildpacks/java-native-image
  paketo-buildpacks/environment-variables             2.1.0          https://github.com/paketo-buildpacks/environment-variables
  paketo-buildpacks/executable-jar                    3.1.1          https://github.com/paketo-buildpacks/executable-jar
  paketo-buildpacks/graalvm                           3.2.2          https://github.com/paketo-buildpacks/graalvm
  paketo-buildpacks/image-labels                      2.0.4          https://github.com/paketo-buildpacks/image-labels
  paketo-buildpacks/maven                             3.1.1          https://github.com/paketo-buildpacks/maven
  paketo-buildpacks/procfile                          2.0.2          https://github.com/paketo-buildpacks/procfile
  paketo-buildpacks/gradle                            3.1.2          https://github.com/paketo-buildpacks/gradle
  paketo-buildpacks/leiningen                         1.1.1          https://github.com/paketo-buildpacks/leiningen
  paketo-buildpacks/sbt                               3.1.1          https://github.com/paketo-buildpacks/sbt
  paketo-buildpacks/spring-boot                       3.2.1          https://github.com/paketo-buildpacks/spring-boot
  paketo-buildpacks/spring-boot-native-image          1.3.2          https://github.com/paketo-buildpacks/spring-boot-native-image
  paketo-buildpacks/java                              3.4.0          https://github.com/paketo-buildpacks/java
  paketo-buildpacks/apache-tomcat                     2.3.0          https://github.com/paketo-buildpacks/apache-tomcat
  paketo-buildpacks/encrypt-at-rest                   2.2.0          https://github.com/paketo-buildpacks/encrypt-at-rest
  paketo-buildpacks/debug                             2.1.1          https://github.com/paketo-buildpacks/debug
  paketo-buildpacks/google-stackdriver                2.6.0          https://github.com/paketo-buildpacks/google-stackdriver
  paketo-buildpacks/azure-application-insights        2.5.0          https://github.com/paketo-buildpacks/azure-application-insights
  paketo-buildpacks/bellsoft-liberica                 4.0.0          https://github.com/paketo-buildpacks/bellsoft-liberica
  paketo-buildpacks/dist-zip                          2.2.0          https://github.com/paketo-buildpacks/dist-zip
  paketo-buildpacks/jmx                               2.1.1          https://github.com/paketo-buildpacks/jmx
  paketo-buildpacks/nginx                             0.0.193        https://github.com/paketo-buildpacks/nginx
  paketo-buildpacks/nodejs                            0.0.7          https://github.com/paketo-buildpacks/nodejs
  paketo-buildpacks/node-start                        0.0.6          https://github.com/paketo-buildpacks/node-start
  paketo-buildpacks/npm-install                       0.2.0          https://github.com/paketo-buildpacks/npm-install
  paketo-buildpacks/npm-start                         0.0.2          https://github.com/paketo-buildpacks/npm-start
  paketo-buildpacks/yarn                              0.0.2          https://github.com/paketo-buildpacks/yarn
  paketo-buildpacks/yarn-install                      0.2.0          https://github.com/paketo-buildpacks/yarn-install
  paketo-buildpacks/yarn-start                        0.0.2          https://github.com/paketo-buildpacks/yarn-start
  paketo-buildpacks/node-engine                       0.1.1          https://github.com/paketo-buildpacks/node-engine
  paketo-buildpacks/go                                0.1.3          https://github.com/paketo-buildpacks/go
  paketo-buildpacks/go-build                          0.0.33         https://github.com/paketo-buildpacks/go-build
  paketo-buildpacks/go-dist                           0.1.2          https://github.com/paketo-buildpacks/go-dist
  paketo-buildpacks/go-mod-vendor                     0.0.166        https://github.com/paketo-buildpacks/go-mod-vendor
  paketo-buildpacks/dep                               0.0.169        https://github.com/paketo-buildpacks/dep
  paketo-buildpacks/dep-ensure                        0.0.30         https://github.com/paketo-buildpacks/dep-ensure

Detection Order:
...
```

---

##### pack info

```
pack report
Pack:
  Version:  0.13.1+git-4134cc6.build-1135
  OS/Arch:  darwin/amd64

Default Lifecycle Version:  0.9.1
```
##### docker info

```
Client:
 Debug Mode: false
 Plugins:
  mutagen: Synchronize files with Docker Desktop (Docker Inc., testing)
  app: Docker Application (Docker Inc., v0.8.0)
  buildx: Build with BuildKit (Docker Inc., v0.3.1-tp-docker)

Server:
 Containers: 10
  Running: 1
  Paused: 0
  Stopped: 9
 Images: 57
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.64GiB
 Name: docker-desktop
 ID: ECK2:RSEE:DKQP:WHTO:UNB3:EALQ:BGNN:DAR2:JFM6:RNQP:XTMS:F7YZ
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 50
  Goroutines: 67
  System Time: 2020-09-29T18:54:06.8789897Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

### PR DESCRIPTION
**Pull request by dwillist**
add a check for buildpack repetitions with the same id and version

Signed-off-by: dwillist <dthornton@vmware.com>

## Summary
Fix a bug due to not checking if we have repeated buildpack `id` & `version` combinations when validating builtpack configuration. 

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Resolves #858

### CODE REVIEW

File: internal/builder/builder.go [hunk 1]
```diff
@@ -408,7 +408,7 @@ func processOrder(buildpacks []dist.BuildpackInfo, order dist.Order) (dist.Order
 			}
 
 			if bpRef.Version == "" {
-				if len(matchingBps) > 1 {
+				if len(matchingBps) > 1 && len(uniqueVersions(matchingBps)) > 1 {
```
Comment (dfreilich) [inline internal/builder/builder.go:411]: nit: Do we need the first check here anymore?

File: internal/builder/builder_test.go [hunk 2]
```diff
@@ -45,8 +45,9 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 		bp1v2          dist.Buildpack
 		bp2v1          dist.Buildpack
 		bpOrder        dist.Buildpack
-		outBuf         bytes.Buffer
-		logger         logging.Logger
+		//bpOrderNoVersion dist.Buildpack
```
Comment (dfreilich) [inline internal/builder/builder_test.go:48]: nit: remove

File: internal/builder/builder_test.go [hunk 3]
```diff
@@ -466,6 +467,28 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 					})
 				})
 
+				when("has repeated buildpacks with the same ID and version", func() {
+					it.Before(func() {
+						subject.AddBuildpack(bp1v1)
+						subject.AddBuildpack(bp1v1)
+					})
+
+					when("order ommits version", func() {
```
Comment (dfreilich) [inline internal/builder/builder_test.go:476]:
```suggestion
					when("order omits version", func() {
```

File: internal/builder/builder_test.go [hunk 4]
```diff
@@ -466,6 +467,28 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
...
+						it("should de-duplicate identical buildpacks", func() {
+							subject.SetOrder(dist.Order{{
```
Comment (dfreilich) [inline internal/builder/builder_test.go:478]: nit: To make the test reflect the bug a bit more, could we have the buildpack be in two separate groups? 

> There is only one version of the given buildpack on the builder image. However, the buildpack is included in two separate groups. My guess is this is the source of the bug.

**Review by dfreilich — APPROVED** [review]
A few nits, but otherwise 👍  from me

looking good

**Review by jromero — APPROVED** [review]

**Pull request merged**